src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
```

Comment:
at this point I use a code like this 

``` 
 const sum = children.reduce((acc, ch) => {
    const mother = people.find(m => m.name === ch.mother);
    return acc + (ch.born - mother.born);
  }, 0);
  ```
  For the same reason for the the other 2 cases, use reduce for make de sum and interation of the data improves the read and the eficiency of the code, for this way you can remove the variable let sum = 0;